src/foraging_gui/Foraging.py

```diff
@@ -3514,15 +3549,40 @@ def log_git_hash():
     '''
         Add a note to the GUI log about the current branch and hash. Assumes the local repo is clean
     '''
+
+    # Get information about python
+    py_version = sys.version
+    logging.info('Python version: {}'.format(py_version))
+    print('Python version: {}'.format(py_version))       
+    if py_version[0:3] != '3.9':
+        logging.error('Incorrect version of python! Should be 3.9, got {}'.format(py_version[0:3]))
+
     try:
+        # Get information about task repository
         git_hash = subprocess.check_output(['git', 'rev-parse', 'HEAD']).decode('ascii').strip()
         git_branch = subprocess.check_output(['git','branch','--show-current']).decode('ascii').strip()
         repo_url = subprocess.check_output(['git', 'remote', 'get-url', 'origin']).decode('ascii').strip()
-        logging.info('Current git commit branch, hash: {}, {}'.format(git_branch,git_hash))
-        return git_hash, git_branch, repo_url
+        repo_dirty = subprocess.check_output(['git','diff-index','--name-only', 'HEAD']).decode('ascii').strip()
```

Comment:
The foraging.bonsai or foraging.bonsai.layout will be changed every session if we save it (the harp COM ports; log folder ). The layout could also be different in different computers (e.g. visualizing the camera could be different). Some changes are automatic. I don't know if there is a good solution for this. I suggest we skip checking dirty for foraging.bonsai and foraging.bonsai.layout or just showing warning (instead of a popup) if we detect dirty for these two files.

Reply:
I don't understand why the `foraging.bonsai` or `foraging.bonsai.layout` files would ever change:
- Changing the harp COM ports should be done through `ForagingSettings.json`, not the bonsai workflow
- Why would you change the log folder?

Reply:
The COM ports in the foraging.bonsai will be overwritten in each computer with different COM ports (one example shown below). 
![image](https://github.com/AllenNeuralDynamics/dynamic-foraging-task/assets/109394934/b8aabdf3-e245-4465-a49c-53c5aafefd8a)

Reply:
But the COM ports are defined in the `Settings_Box.csv` files. Why do you change them in the bonsai workflow?

Reply:
We don't need to change the foraging.bonsai, but the bonsai changes it by default. If we save the bonsai workflow, these changes will be saved. We don't see the dirty information in the behavior rig because we don't save the changed bonsai workflow. 

If we hope to use a different layout to visualize the video capture and hope to save the layout, these changes in the foraging.bonsai (including com ports and log folder path) will also be saved automatically. 

I also suggest we save the list of dirty files.

Reply:
@XX-Yin I've updated the code to save which files were dirty, both in the log, and in the `json`
- I don't understand why you would change the layout of the video capture. 
- If you change the com ports and the log folder path, this will be done through the `.csv` file, and wont result in the workflow changing.

Reply:
Note that the user can always just press "ok" to continue running with untracked local changes

Reply:
It's ok for me to click extra "ok" button if there are some changes.